tests/test_storage.py

```diff
@@ -25,87 +25,128 @@ def _make_local_files(files):
+    zg = access(store, mode="w")
+    zg["foo"] = data
+    assert zarr.open(DEFAULT_ZARR_STORE(store), mode="a") == zg
 
 
-def test_accessing_group_zarr_n5():
-    store = tempfile.mkdtemp(suffix='.n5')
+def test_access_group_zarr_n5():
+    store = tempfile.mkdtemp(suffix=".n5")
     atexit.register(shutil.rmtree, store)
-    data = np.zeros(100, dtype='uint8') + 42
-    zg = zarr.open(store, mode='a')
-    zg.attrs.update({'foo': 'bar'})
-    zg['foo'] = data
+    data = np.zeros(100, dtype="uint8") + 42
+    zg = zarr.open(store, mode="a")
+    zg.attrs.update({"foo": "bar"})
+    zg["foo"] = data
 
-    assert dict(access(store, mode='r').attrs) == {'foo': 'bar'}
-    assert np.array_equal(access(store, mode='r')['foo'][:], data)
+    assert dict(access(store, mode="r").attrs) == {"foo": "bar"}
+    assert np.array_equal(access(store, mode="r")["foo"][:], data)
 
 
-def test_accessing_precomputed():
-    store = tempfile.mkdtemp(suffix='.precomputed')
+def test_access_precomputed():
+    store = tempfile.mkdtemp(suffix=".precomputed")
     atexit.register(shutil.rmtree, store)
-    key = 's0'
-    data = np.random.randint(0, 255, size=(10, 10, 10), dtype='uint8')
+    key = "s0"
+    data = np.random.randint(0, 255, size=(10, 10, 10), dtype="uint8")
     resolution = [1.0, 2.0, 3.0]
     chunks = (2,) * data.ndim
 
-    arr_out = access_precomputed(store,
-                                 key,
-                                 mode='w',
-                                 array_type='image',
-                                 dtype='uint8',
-                                 shape=data.shape,
-                                 num_channels=1,
-                                 resolution=resolution,
-                                 encoding='raw',
-                                 chunks=chunks)
-    arr_out[:] = data.reshape(*data.shape,1)
-
-    arr_in = access_precomputed(store, key, mode='r')
+    arr_out = access_precomputed(
+        store,
+        key,
+        mode="w",
+        array_type="image",
+        dtype="uint8",
+        shape=data.shape,
+        num_channels=1,
+        resolution=resolution,
+        encoding="raw",
+        chunks=chunks,
+    )
+    arr_out[:] = data.reshape(*data.shape, 1)
+
+    arr_in = access_precomputed(store, key, mode="r")
     assert np.all(np.array(arr_in[:]) == data.reshape(*data.shape, 1))
 
     darr = precomputed_to_dask(os.path.join(store, key), chunks=(2, 2, 2))
     assert (darr.compute() == data).all
 
 
+def test_access_array_h5():
+    key = "s0"
+    data = np.random.randint(0, 255, size=(10, 10, 10), dtype="uint8")
+    attrs = {"resolution": "1000"}
+    chunks = (2,) * data.ndim
+
+    with tempfile.TemporaryFile(suffix=".h5") as store:
+        arr = access_h5(store, key, chunks=chunks, data=data, attrs=attrs, mode="w")
+        assert dict(arr.attrs) == attrs
+        assert np.array_equal(arr[:], data)
+        arr.file.close()
```

Comment:
Add a test to see if you can add multiple datasets to a single file.

Reply:
I'm not sure what that tests for -- how would it be possible to add 1, but not multiple datasets to a file?

Reply:
Perhaps there is some usage mode where an internal object handle (such as a dataset handle) leaked and the file did not actually close. On the next access, you would not be able to reopen the file for writing.

Since the access pattern you're using here involves triggering global operating system file locks and internal HDF5 locks at relatively high frequency, it would be good to test that this usage pattern does not cause issues.

The `h5py` interfaces to closing a file are unfortunately quite expensive relative to the C code:

`h5py.File.close()`:
https://github.com/h5py/h5py/blob/e0f960d93598ca1189510a973de94044bb7addeb/h5py/_hl/files.py#L542-L555

`h5py.FileID._close_open_objects()`
https://github.com/h5py/h5py/blob/3b75dc4882acbc9cf8cd6270388a4a234471b5bc/h5py/h5f.pyx#L345-L362

Simple dataset read / write in C:
https://raw.githubusercontent.com/HDFGroup/hdf5/develop/examples/h5_rdwt.c